OBAKit/Helpers/OBAStopIconFactory.m

```diff
@@ -16,38 +16,298 @@
 
 #import <OBAKit/OBAStopIconFactory.h>
 #import <OBAKit/OBARouteV2.h>
+#import <OBAKit/OBACanvasView.h>
 
 @implementation OBAStopIconFactory
 
+// . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
+
+// The construction is a large transparent image that contains the icon in its
+// center, with room on all sides for optional direction chevrons. This way, the
+// the center of the icon image remains consistently placed on the map regardless
+// of where the direction chevron is placed.
+//
+// The icon is the rounded rect describing the stop, with the glyph of the
+// transport type and the label text for STOP.
+
+//    +-----------------+ ---
+//    |    <chevron>    |  .
+//    |   +---------+   |  .
+//    |   |  icon   |   |  .
+//    |   | +-----+ |   |  .
+//    |<c>| |glyph| |<c>|  Overall View
+//    |   | | --- | |   |  .
+//    |   | | txt | |   |  .
+//    |   | +-----+ |   |  .
+//    |   +---------+   |  .
+//    |    <chevron>    |  .
+//    +-----------------+ ---
+
+// These constants define the geometry of the stop icon, the glyph and text
+// placement, and the size and position of the direction chevron.
+
+#define IconCornerRadiusPercentage  0.1f    // Radius of icon corner, relative to overall size
+#define StrokeWidthPercentage       0.038f  // Width of stroke for all borders, relative to overall size
+#define IconInsetPercentage         0.21f   // Amount of padding to inset the icon (leaving room for chevron), relative to overall size
+#define GlyphAreaPercentage         0.7f    // Amount of icon height reserved for glyph
+#define TextAreaPercentage          (1.0f - GlyphAreaPercentage) // Amount of icon height reserved for text
+#define GlyphHeightPercentage       0.6f    // Height of glyph, relative to icon height
+#define GlyphInsetRelativeToStroke  1.5f    // Padding around glyph, relative to stroke width
+#define ChevronWidthPercentage      0.25f   // Width of chevron, relative to overall size (assuming N chevron)
+#define ChevronHeightPercentage     0.15f   // Height of chevron, relative to overall size (assuming N chevron)
+
+// For the simple getIconForStop call, here are the default dimensions
+#define StopIconDefaultWidth        54
+#define StopIconDefaultHeight       54
+
+// . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
+
+static NSDictionary *rotationAngles;
+static UIColor *strokeColor;
+static UIColor *iconBackgroundColor;
+static UIColor *textBackgroundColor;
+static UIColor *chevronFillColor;
+static UIColor *textColor;
+static NSString *stopLabelFaceName;
+static NSString *stopLabelText;
+static NSCache *iconCache;
+
++(void)initialize {
+    iconCache = [[NSCache alloc] init];
+
+    stopLabelText = NSLocalizedString(@"STOP",);
+
+    // I chose "Helvetica" because it's the official typeface of the MTA
+    stopLabelFaceName = @"Helvetica Bold";
```

Comment:
Ok, `stopLabelFaceName = [OBATheme titleFont].familyName;` it is. Had to move the initializers out of `+(void)initialize` and into a dispatch once block since it calls UIKit code.